### ISSUE TITLE
Enhancement: [naming-convention] alias matches existing identifier

### ISSUE THREAD

**Issue by abaran30** [typescript-eslint/typescript-eslint#7439]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention

### Description

I propose that the `naming-convention` rule should allow for some option, e.g. `aliasMatchesExisting`, where local-scope aliases use the format of the existing identifier.

### Fail

```typescript
const CAPACITY = 5;

class Teapot {
  readonly capacity = CAPACITY;
}
```

### Pass

```typescript
const CAPACITY = 5;

class Teapot {
  readonly CAPACITY = CAPACITY;
}
```

### Additional Info

The idea of this enhancement comes from [TS Style Guide - Aliases](https://ts.dev/style/#aliases).

I'd be more than happy to contribute the implementation of this idea. However, I first wanted to see if this idea makes sense as an enhancement to `naming-convention` or perhaps a new rule. Please let me know what you think; happy to discuss. 🙂

**Comment by bradzacher** [comment]
I don't think this is a style that always applies and thus isn't something that you should lint for.
Instead I think the rules should be broader and more consistent - i.e. Matching our other rules.

For example if one wants readonly class members to be upper case then that's something that should be done always - and not just because the value being assigned was in a specific case.

For example:
```ts
const CAPACITY = 5;

class Teapot {
  readonly CAPACITY = CAPACITY;
  readonly constant = 5;
}
```

This code is inconsistent and thus stylistically bad. Why should one be upper-cased vs the other? Why should you be forced to inherit some library's casing rules? eg if a library exports `CrAzY_cAsE` - as per the style guide you must use that - which is CrAzY!

Whilst they have a catchy URL - I don't think that their style guide is applicable to the broader community and thus I am going to reject this proposal for this project.
